### PR TITLE
[nodejs] Add chart titles as part of returned image

### PR DESCRIPTION
**Pull request by chejennifer**
Screenshot 2024-01-11 at 3 15 33 PM
Screenshot 2024-01-11 at 3 16 01 PM
Screenshot 2024-01-11 at 3 14 42 PM

Note: Ideally would like to use larger font for the chart title, but the hack for getting bbox to work on node server depends on the text within a chart to be all the same size ..

### CODE REVIEW

File: static/js/chart/draw_utils.ts [hunk 1]
```diff
@@ -640,3 +640,26 @@ export function updateXAxis(
...
     );
 }
+
+/**
+ * Adds a chart title to the top of the svg and returns the height of the chart
+ * title that was added.
+ * @param svg svg to add title to
+ * @param titleText text of the title to add
+ * @param width width of the svg
+ */
```
Comment (juliawu) [inline static/js/chart/draw_utils.ts:650]: nit: add a `@returns`

File: static/js/chart/draw_scatter.ts [hunk 2]
```diff
@@ -853,13 +849,18 @@ export function drawScatter(
   const xMinMax = d3.extent(Object.values(points), (point) => point.xVal);
   const yMinMax = d3.extent(Object.values(points), (point) => point.yVal);
 
-  let height = properties.height - MARGINS.top - MARGINS.bottom;
+  let marginTop = MARGINS.top;
```
Comment (juliawu) [inline static/js/chart/draw_scatter.ts:852]: Discussion: should we rename MARGINS to MARGIN to match the naming used in the other files?

**Review by juliawu — APPROVED** [review]
Very nice!

File: static/js/chart/draw_bar.ts [hunk 3]
```diff
@@ -282,14 +283,19 @@ export function drawStackBarChart(
   const xAxis = svg.append("g").attr("class", "x axis");
   const tempYAxis = svg.append("g");
 
+  let marginTop = MARGIN.top;
+  if (options?.title) {
+    marginTop += addChartTitle(svg, options?.title, chartWidth);
```
Comment (dwnoble) [inline static/js/chart/draw_bar.ts:288]: Nit: remove `?` here? (& in other instances)

**Review by dwnoble — APPROVED** [review]
Nice!

File: static/js/chart/draw_utils.ts (hunk 1 above)
Comment (chejennifer) [reply static/js/chart/draw_utils.ts:651]: done

File: static/js/chart/draw_scatter.ts (hunk 2 above)
Comment (chejennifer) [reply static/js/chart/draw_scatter.ts]: i like that, done

File: static/js/chart/draw_bar.ts (hunk 3 above)
Comment (chejennifer) [reply static/js/chart/draw_bar.ts]: done

**Pull request merged**